client/fingerprint_manager.go

```diff
@@ -86,25 +111,93 @@ func (fm *FingerprintManager) setupDrivers(drivers []string) error {
 
 		p, period := d.Periodic()
 		if p {
-			go fm.run(d, period, name)
+			go fm.runFingerprint(d, period, name)
+		}
+
+		// We should only run the health check task if the driver is detected
+		// Note that if the driver is detected later in a periodic health check,
+		// this won't automateically trigger the periodic health check.
+		if hc, ok := d.(fingerprint.HealthCheck); ok && detected {
+			req := &cstructs.HealthCheckIntervalRequest{}
+			resp := &cstructs.HealthCheckIntervalResponse{}
+			hc.GetHealthCheckInterval(req, resp)
+			if resp.Eligible {
+				go fm.runHealthCheck(hc, resp.Period, name)
+			}
 		}
 	}
 
 	fm.logger.Printf("[DEBUG] client.fingerprint_manager: detected drivers %v", availDrivers)
 	return nil
 }
 
+// fingerprintDriver is a temporary solution to move towards DriverInfo and
+// away from annotating a node's attributes to demonstrate support for a
+// particular driver. Takes the FingerprintResponse and converts it to the
+// proper DriverInfo update and then sets the prefix attributes as well
+func (fm *FingerprintManager) fingerprintDriver(name string, f fingerprint.Fingerprint) (bool, error) {
```

Comment:
This is only being called the first time. Periodic, health checking drivers should always call this. Where "this" is the merging logic into the driverinfo